cmd/jag/commands/firmware.go

```diff
@@ -150,48 +139,51 @@ func BuildFirmwareImage(ctx context.Context, id string, name string, wifiSSID st
 		return nil, err
 	}
 
-	configFile, err := os.CreateTemp("", "*.json")
+	envelopeInput := filepath.Join(esp32BinPath, "firmware.envelope")
+	envelope, err := os.CreateTemp("", "*.envelope")
 	if err != nil {
 		return nil, err
 	}
-	defer os.Remove(configFile.Name())
-
-	var config binaryConfig
-	config.ID = id
-	config.Name = name
-	config.Wifi.SSID = wifiSSID
-	config.Wifi.Password = wifiPassword
-	if err := json.NewEncoder(configFile).Encode(config); err != nil {
-		configFile.Close()
-		return nil, err
-	}
-	configFile.Close()
-
-	toitBin := filepath.Join(esp32BinPath, "toit.bin")
+	defer envelope.Close()
 
 	binTmpFile, err := os.CreateTemp("", "*.bin")
 	if err != nil {
 		return nil, err
 	}
 
-	binFile, err := os.Open(toitBin)
+	jaguarSnapshot, err := directory.GetJaguarSnapshotPath()
 	if err != nil {
-		binTmpFile.Close()
 		return nil, err
 	}
 
-	_, err = io.Copy(binTmpFile, binFile)
-	binTmpFile.Close()
-	binFile.Close()
-	if err != nil {
+	// TODO(kasper): Can we generate this in a nicer way?
+	wifiConfig := "{ \"wifi.ssid\": \"" + wifiSSID + "\", \"wifi.password\": \"" + wifiPassword + "\" }"
```

Comment:
Maybe?
```suggestion
	wifiConfig := fmt.Sprintf("{ \"wifi.ssid\": \"%s\", \"wifi.password\": \"%s"\" }", wifiSSID, wifiPassword)
```
Alternatively, obviously, using JSON.

Reply:
I'll probably change it to map -> json.stringify soon.